test/Integration/modules/edgecore/eventbus/eventbus_test.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"database/sql"
+	"encoding/json"
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	_ "github.com/mattn/go-sqlite3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/kubeedge/kubeedge/test/Integration/utils/common"
+)
+
+//DeviceUpdate device update
+type DeviceUpdate struct {
+	State string `json:"state,omitempty"`
+}
+
+//Function to access the edge_core DB and return the device state.
+func checkDeviceState(deviceId string) string {
+
+	var id []uint8
+	var name string
+	var Description string
+	var State string
+	var LastOnline []uint8
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		common.InfoV6("Failed to get PWD: %v", err)
+		os.Exit(1)
+	}
+	destpath := filepath.Join(pwd, "../../edge.db")
+	db, err := sql.Open("sqlite3", destpath)
+	if err != nil {
+		common.InfoV6("Open Sqlite DB failed :%v", err)
+	}
+	defer db.Close()
+	state, _ := db.Query("SELECT * FROM device", 0)
+	if err != nil {
+		common.InfoV6("Query Sqlite DB failed: %v", err)
+	}
+	defer state.Close()
+	for state.Next() {
+		err = state.Scan(&id, &name, &Description, &State, &LastOnline)
+		if err != nil {
+			common.InfoV6("Failed to scan DB rows: %v", err)
+		}
+
+		if string(id) == deviceId {
+			break
+		}
+	}
+	return State
+}
+
+//Run Test cases
+var _ = Describe("Event BUS Testing", func() {
+	var Token_client Token
+	var ClientOpts *MQTT.ClientOptions
+	var Client MQTT.Client
+	Context("Individual Create", func() {
+		BeforeEach(func() {
+			ClientOpts = HubclientInit(ctx.Cfg.MqttEndpoint, ClientID, "", "")
+			Client = MQTT.NewClient(ClientOpts)
+			if Token_client = Client.Connect(); Token_client.Wait() && Token_client.Error() != nil {
+				common.InfoV6("client.Connect() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Connect success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+
+		})
+		AfterEach(func() {
+			common.PrintTestcaseNameandStatus()
+		})
+
+		It("TC_TEST_EBUS_1 :Sending Messge on SYS/dis/upload_records Topic", func() {
+
+			if Token_client = Client.Publish(Upload_record_to_cloud, 0, false, "messagetoUpload_record_to_cloud"); Token_client.Wait() && Token_client.Error() != nil {
```

Comment:
This test just publishes onto a topic for sending data. This is only testing if an mqtt publish works fine.
What happens after the publish ? We need to test whether event bus received the event correctly and whether it forwarded it to edgehub.

Reply:
@rohitsardesai83 , for verfication of publish message is received by eventbus or not is handled in seperate test cases (handled via device testing), as eventbus doesn't response back to application/user, the above test cases you have mentioned only intended for broker connection validation.